lib/getUniqueMatch.js

```diff
@@ -2,19 +2,25 @@
 var Cesium = require('cesium');
 var defined = Cesium.defined;
 
+module.exports = getUniqueMatch;
+
 /** Generic regex search over array of String. (Requires global regex)
```

Comment:
Please update the description to be a little more clear, also verify the input data rather than letting an infinite loop happy, you should be able to tell if it's global or not via the https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/global property.